### DISCUSSION TITLE
multiple select with chips as display value

### DISCUSSION THREAD

**Discussion by patrickwasp** [zauberzeug/nicegui#817] [Ideas / Feature Requests]
The current implementation of select appears to only allow a single selection. https://nicegui.io/documentation/select

It would be nice to be able to select multiple values and have them displayed as chips

Additional documentation about this component is here: 
 - https://quasar.dev/vue-components/select#example--chips-as-display-value

**Comment by falkoschindler** [comment]
That would be a great feature, @patrickwasp!
Displaying the selected value with chips is already possible (but rather useless for single selection):
```py
ui.select(['red', 'green', 'blue'], label='Select a color').props('use-chips')
```

Multi-selection is a bit trickier, because `ui.select` derives from `ChoiceElement` (like `ui.radio` and `ui.toggle`), which is built for single selection only. Maybe we can extend the underlying model to support multiple values. Otherwise we could introduce a dedicated `ui.multi_select` element.

**Comment by rodja** [comment]
The feature was just asked for on Discord again: https://discord.com/channels/1089836369431498784/1109256613375778886

**Comment by ghost** [comment]
I hope the multi select element will be available soon

**Reply by MJAGO71** [reply]
I'm also looking forward to this element, I'm migrating a plottly dash application that has this functionality and I'm having to resolve this in the database query.

**Comment by falkoschindler** [comment]
Ok, we should definitely look into this feature. As a workaround you can build a simple multi selection like this:

```py
select = ui.element('q-select').props('multiple use-chips')
select._props['options'] = ['Alice', 'Bob', 'Carol']

def handle_change(msg: dict) -> None:
    select._props.update({'model-value': msg['args']})
    select.update()
    ui.notify(f'Value changed to {msg["args"]}')

select.on('update:model-value', handle_change)
```

**Comment by falkoschindler** [comment]
I just created PR https://github.com/zauberzeug/nicegui/pull/995 introducing multi-select support for `ui.select`. What do you think?

**Comment by falkoschindler** [comment]
I'll close this feature request as completed in version 1.2.18. 🎉
